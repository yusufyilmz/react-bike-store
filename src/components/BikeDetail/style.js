


import styled from 'styled-components';


export const Div = styled.div`
    overflow: scroll;
`;

export const H5 = styled.h5`
    margin-left: 10px;
`;

export const Ul = styled.ul`
    overflow: scroll;
    padding: 10px;
    padding-bottom: 20px;`;




