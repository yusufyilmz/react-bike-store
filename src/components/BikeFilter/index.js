


import React from 'react';
import Input from '../Input';
import { Form } from './style';
import BikeFilterContainer from '../../containers/bikeFilterContainer'

const UserInput = () => {
    return (
        <BikeFilterContainer>
            {({description, dateFrom, dateTo, changeDescription, changeDateFrom, changeDateTo, filterStolenBikes }) => (
                <Form onSubmit={filterStolenBikes}>
                    <Input
                        type="text"
                        name="company"
                        placeHolder="Search Case and descriptions"
                        width={230}
                        value={description}
                        onChange={changeDescription}
                        required={true} />
                    <Input
                        type="date"
                        name="startDate"
                        text="from: "
                        value={dateFrom}
                        onChange={changeDateFrom}
                        required={true} />
                    <Input
                        type="date"
                        name="endDate"
                        text="to: "
                        value={dateTo}
                        onChange={changeDateTo}
                        required={true} />
                    <Input type="submit" value="Filter" />
                </Form>
            )}
        </BikeFilterContainer>

    );
}

export default UserInput;