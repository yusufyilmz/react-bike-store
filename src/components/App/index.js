import React, { Component } from 'react';
import { Container, H1, Main } from './style';
import {Header} from '../Header';
import BikeStore from '../BikeStore';

class App extends Component {

    render() {
        return (
            <Main>
                <Container>
                    <Header />
                    <BikeStore/>
                </Container>
            </Main>
        );
    }
}

export default App;
