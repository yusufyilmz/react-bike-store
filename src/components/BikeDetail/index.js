






import React from 'react';
import Bike from '../Bike';
import {Div, H5, Ul} from './style';


export const BikeDetail = ({ title, description, occured, sourceUrl, address }) => {

    return (
        <Div>
            <a href={sourceUrl}>
                <H5>
                    <strong> {title} </strong>
                </H5>
            </a>
            <Ul>
                <li>
                    <span>{description}</span> 
                </li>
                <br/>
                <li>
                    <span>{occured} - {address} </span> 
                </li>
            </Ul>
        </Div>
    );
}






// <div class="bike-information multi-attr-lists">
//     <h5 class="title-link">
//         <a href="https://bikeindex.org/bikes/486269">
//             <strong>
//                 2015 Centurion
//       </strong>
//             Backfire Race 1000.27 metallic-grau/schwarz-lime 51 cm
//       </a>
//     </h5>
//     <ul class="attr-list">
//         <li><span class="attr-title">Serial</span>AW470157</li>
//         <li><span class="attr-title">Primary colors</span>Silver, gray or bare metal and Green</li>
//     </ul>
//     <ul class="attr-list">
//         <li>
//             <span class="attr-title">
//                 Stolen
//          </span>
//             <span class="">Dec 3rd, 2am</span>
//         </li>
//         <li><span class="attr-title">Location</span>Munich - DE</li>
//     </ul>
// </div>