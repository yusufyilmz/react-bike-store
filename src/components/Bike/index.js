

import React from 'react';
import { Li } from './style';
import { BikeImage } from '../BikeImage'
import { BikeDetail } from '../BikeDetail'


export const Bike = ({onClick, ...bike }) => {
    return (
        <Li onClick={() => onClick(bike)}>
            <BikeImage
                imageUrl={bike.imageUrl}
                imageAlt={bike.title} />
            <BikeDetail
                title={bike.title}
                sourceUrl={bike.sourceUrl}
                address={bike.address}
                occured={bike.occurred}
                description={bike.description} />
        </Li>
    );
}
export default Bike;
