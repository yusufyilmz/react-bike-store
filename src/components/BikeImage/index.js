

import React from 'react';
import Bike from '../Bike';
import { Img, A } from './style';



export const BikeImage = ({ url, imageUrl, imageAlt }) => {

    const image = imageUrl ? imageUrl :  "http://eni-learning.com/php-custom/img/no-image.png";
    return (
        <A href={url}>
            <Img
                alt={imageAlt}
                src={image} />
        </A>
    );
}

