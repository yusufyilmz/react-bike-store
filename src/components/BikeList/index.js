

import React from 'react';
import Bike from '../Bike';
import { ListWrapper } from './style';



const BikeList = ({ bikes }) => {

    return (
        <ListWrapper>
            {bikes.map(bike => {
                return <Bike {...bike} />
            })}
        </ListWrapper>
    );
}

export default BikeList;
