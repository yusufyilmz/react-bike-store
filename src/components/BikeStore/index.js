

import React, { Component, Fragment } from 'react';
import BikeStoreContainer from '../../containers/bikeStoreContainer';
import Bike from '../Bike';

import { ListWrapper } from './style';

class BikeStore extends Component {

    render() {
        return (

            <BikeStoreContainer>
                {({ bikes }) => (

                )}

            </BikeStoreContainer>
        );
    }
}

export default BikeStore;
