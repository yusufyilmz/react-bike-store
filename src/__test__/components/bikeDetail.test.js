

import React from 'react';
import {BikeDetail} from '../../components/BikeDetail';
import { shallow } from 'enzyme'


describe('BikeDetail component', () => {

    let value, wrapper;

    beforeEach(() => {
        value = 'test';
        wrapper = shallow(
            <BikeDetail title={value} />
        );
    });

    test('renders element correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders strong correctly', () => {
        expect(wrapper.find('strong').text()).toContain(value);
    });
})
