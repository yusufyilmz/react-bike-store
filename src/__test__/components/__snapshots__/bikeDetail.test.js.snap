// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeDetail component renders element correctly 1`] = `
<styled.div>
  <a
    href="#"
  >
    <styled.h5>
      <strong>
         
        test
         
      </strong>
    </styled.h5>
  </a>
  <styled.ul>
    <li>
      <span />
    </li>
    <br />
    <li>
      <span>
        Invalid Date
         - 
         
      </span>
    </li>
  </styled.ul>
</styled.div>
`;
